documentation/src/main/asciidoc/mapstruct-reference-guide.asciidoc

```diff
@@ -1974,3 +1974,139 @@ The order in which the selected methods are applied is roughly determined by the
+        this.name = name;
+    }
+----
+====
+
+We want `GolfPlayer` to be mapped to a target object `GolfPlayerDto` similar like we 'always' do this:
+
+.Source object with fluent API.
+====
+[source, java, linenums]
+[subs="verbatim,attributes"]
+----
+@Mapper
+public interface GolfPlayerMapper {
+
+    GolfPlayerMapper INSTANCE = Mappers.getMapper( GolfPlayerMapper.class );
+
+    GolfPlayerDto golfPlayerToDto(GolfPlayer player);
+}
+----
+====
+
+This can be achieved with implementing a ServiceProviderInterface (SPI). Here's an implemented `org.mapstruct.ap.spi.AccessorNamingStrategy`:
+
+.CustomAccessorNamingStrategy
+====
+[source, java, linenums]
+[subs="verbatim,attributes"]
+----
+public class CustomAccessorNamingStrategy implements AccessorNamingStrategy {
+
+    @Override
+    public MethodType getMethodType(ExecutableElement method) {
+        if ( isGetterMethod( method ) ) {
+            return MethodType.GETTER;
+        }
+        else if ( isSetterMethod( method ) ) {
+            return MethodType.SETTER;
+        }
+        else if ( isAdderMethod( method ) ) {
+            return MethodType.ADDER;
+        }
+        else {
+            return MethodType.OTHER;
+        }
+    }
+
+    private boolean isGetterMethod(ExecutableElement method) {
+        return method.getReturnType().getKind() != TypeKind.VOID;
+    }
+
+    public boolean isSetterMethod(ExecutableElement method) {
+        String methodName = method.getSimpleName().toString();
+
+        return methodName.startsWith( "with" ) && methodName.length() > 4;
+    }
+
+    public boolean isAdderMethod(ExecutableElement method) {
+        String methodName = method.getSimpleName().toString();
+        return methodName.startsWith( "add" ) && methodName.length() > 3;
+    }
+
+    @Override
+    public String getPropertyName(ExecutableElement getterOrSetterMethod) {
+        String methodName = getterOrSetterMethod.getSimpleName().toString();
+        return Introspector.decapitalize( methodName.startsWith( "with" ) ? methodName.substring(  4 ) : methodName );
+    }
+
+    @Override
+    public String getElementName(ExecutableElement adderMethod) {
+        String methodName = adderMethod.getSimpleName().toString();
+        return Introspector.decapitalize( methodName.substring( 3 ) );
+    }
+
+    @Override
+    public String getCollectionGetterName(String property) {
+        return property.substring( 0, 1 ).toUpperCase() + property.substring( 1 );
+    }
+}
+----
+====
+Adding an SPI is as simple as adding a file named `org.mapstruct.ap.spi.AccessorNamingStrategy` with the simple content `org.mapstruct.example.CustomAccessorNamingStrategy` (FQN) to the `META-INF/services/` folder.
+
+This as a rudimentary SPI which of course can be improved. But the uptake is that MapStruct identifies a target write accessor (setter) as starting with "with" and a target read accessor with no particular prefix. The generated code will use that 'setter' and 'getter' definition.
```

Comment:
I'd remove this whole line and add the description of what's done directly as comments to the code sample, making it easier for the reader. If the code example extends the default implementation it should be short and snappy enough to get the idea. The interested reader then should be able to get the idea of it and continue in the code or the javadocs to see how their specific problem can be solved.